data_model/src/permission/validator.rs

```diff
@@ -1,133 +1,31 @@
 //! Structures, traits and impls related to *runtime* `Validator`s.
-//!
-//! # Note
-//!
-//! Currently Iroha 2 has only builtin validators (see `core/src/smartcontracts/permissions`).
-//! They are partly using API from this module.
-//! In the future they will be replaced with *runtime validators* that use WASM.
-//! The architecture of the new validators is quite different from the old ones.
-//! That's why some parts of this module may not be used anywhere yet.
-use iroha_data_model_derive::IdEqOrdHash;
-use iroha_macro::FromVariant;
 
 use super::*;
 use crate::{
-    account::Account,
-    expression::Expression,
     isi::Instruction,
     model,
     query::QueryBox,
     transaction::{SignedTransaction, WasmSmartContract},
-    ParseError,
+    FromVariant,
 };
 
 model! {
-    /// Identification of a [`Validator`].
-    ///
-    /// Consists of Validator's name and account (authority) id
-    #[derive(Debug, Display, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Constructor, Getters, Decode, Encode, DeserializeFromStr, SerializeDisplay, IntoSchema)]
-    #[display(fmt = "{name}%{owned_by}")]
-    #[getset(get = "pub")]
-    #[ffi_type]
-    pub struct Id {
-        /// Name given to validator by its creator.
-        pub name: Name,
-        /// Account that owns the validator.
-        pub owned_by: <Account as Identifiable>::Id,
-    }
-
     /// Permission validator that checks if an operation satisfies some conditions.
     ///
     /// Can be used with things like [`Transaction`]s,
     /// [`Instruction`]s, etc.
-    #[derive(Debug, Display, Clone, IdEqOrdHash, Constructor, Getters, Decode, Encode, Deserialize, Serialize, IntoSchema)]
+    #[derive(Debug, Clone, PartialEq, Eq, Hash, Constructor, Getters, Decode, Encode, Deserialize, Serialize, IntoSchema)]
     #[allow(clippy::multiple_inherent_impl)]
-    #[display(fmt = "{id}")]
-    #[ffi_type]
+    #[ffi_type(opaque)]
```

Comment:
```suggestion
    #[ffi_type(unsafe {robust})]
```

Reply:
If wish Rust had docs for derive attributes, so that I can read exactly what `ffi_type` does

Reply:
I'll try to improve it